icepyx/core/spatial.py

```diff
@@ -364,7 +410,22 @@ def validate_polygon_file(spatial_extent):
 
 
 class Spatial:
-    def __init__(self, spatial_extent, **kwarg):
+    _ext_type: ExtentType
+    _spatial_ext: list[float]
+    _geom_file: Optional[str]
+
+    def __init__(
+        self,
+        spatial_extent: Union[
+            str,  # Filepath
+            list[str],  # Bounding box or polygon
```

Comment:
```suggestion
```
List of strings isn't an option we handle.

Reply:
Ok, I think the docstrings and type annotations here are misleading and I can see some places where the list of strings would not work.

It looks like maybe this was something that was supported in the past, or had been planned but not all-the-way completed? In any case, I think I'll create another branch to address these issues.

Reply:
The changes were not as extensive as I thought they might be, so I made a new commit with the change to remove typing of `list[str]`: b035695

Reply:
> Ok, I think the docstrings and type annotations here are misleading and I can see some places where the list of strings would not work.

This is reminding me that there were a few outstanding updates to the Spatial module docstrings that never got completed by the student who created the module... since this PR is adding the typing I definitely want to make sure we get that right (and was probably not looking at the docstrings when I checked the types)! Thanks for fixing the docstrings to match.